pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -213,6 +215,16 @@ public List<StarTreeV2> getStarTrees() {
     return _starTreeIndexContainer != null ? _starTreeIndexContainer.getStarTrees() : null;
   }
 
+  @Nullable
+  public void setValidDocSnapshots(ImmutableRoaringBitmap validDocSnapshots) {
```

Comment:
I feel a easier way to manage the snapshot is to add 3 methods into this class:
- `@Nullable ImmutableRoaringBitmap loadValidDocIdsSnapshot()`
- `void persistValidDocIdsSnapshot()`
- `void deleteValidDocIdsSnapshot()`

One scenario we want to handle is that when snapshot is enabled then disabled, we should be able to delete the existing snapshots.

Reply:
gotcha, thanks for pointing out, let me address this.